core/src/main/java/kafka/server/TierStateMachine.java

```diff
@@ -54,5 +180,109 @@ PartitionFetchState start(TopicPartition topicPartition,
      * @return the new PartitionFetchState if the tier state machine was advanced, otherwise, return the currentFetchState
      */
     Optional<PartitionFetchState> maybeAdvanceState(TopicPartition topicPartition,
-                                                    PartitionFetchState currentFetchState);
+                                                    PartitionFetchState currentFetchState) {
+        // This is currently a no-op but will be used for implementing async tiering logic in KAFKA-13560.
```

Comment:
KAFKA-13560 seems to stale for a while. When (and if?) it gets implemented, it would be simple to add this async placeholder. In case it doesn't, or even if it simply takes a long time, I think it would make sense to keep the interface and definiton in this class free from that future intention. We could drop this unused method, rename `start()` to something like `nextState()`, and move any documentation here that may be relevant in the future to the JIRA. What do you think?

Reply:
I think it makes sense we remove this unused placeholder first, and add it back when implemented.